pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/LookupHelper.java

```diff
@@ -576,6 +593,52 @@ private void buildMessage(
         }
     }
 
+    /**
+     * Extract the set of fields that are comparable for a given message.
+     * @param msgDef The message definition to get comparable fields for
+     * @return a list of field names that are comparable
+     */
+    static List<String> extractComparableFields(final MessageDefContext msgDef) {
+        if (msgDef.optionComment() == null || msgDef.optionComment().getText() == null) {
+            return emptyList();
+        }
+        final var matcher = OPTION_COMMENT.matcher(msgDef.optionComment().getText());
+        if (matcher.find()) {
+            final String optionName = matcher.group(1);
+            final String optionValue = matcher.group(2);
+            if (optionName.equals(PBJ_COMPARABLE_OPTION_NAME)) {
+                final Set<String> regularFieldNames = msgDef.messageBody().messageElement().stream()
+                        .filter(v -> v.field() != null)
+                        .filter(v -> {
+                            if (v.field().REPEATED() != null) {
+                                throw new IllegalArgumentException(("Field `%s` specified in `%s` option is repeated. " +
```

Comment:
I'm not sure if this stream goes over fields listed in the annotation. I believe this stream iterates over all the fields declared in the model. So this `throw` will activate even when the annotation doesn't request a repeated field to be included in the compareTo.

I suggest to filter by the values from the `optionValue` first, and only then perform other validations.

Note that you'd still need to additionally check that optionValue doesn't list any non-existing fields.

Reply:
> So this throw will activate even when the annotation doesn't request a repeated field to be included in the compareTo.

Good catch. I will add a test for that.

Reply:
Done